src/secureCreatives.js

```diff
@@ -63,6 +63,11 @@ export function receiveMessage(ev) {
       } else if (data.action === 'allAssetRequest') {
         const message = getAllAssetsMessage(data, adObject);
         ev.source.postMessage(JSON.stringify(message), ev.origin);
+        // if there is an ortb object inside native, puc won't send postMessage to trigger impression tracker, in that case marking bid as winning
```

Comment:
I think it'd be better to move this to the top, if any message with an `adId` comes in, it means that `adId` has won the auction, no? you'll need some logic to avoid duplicating the event but IMO that's better than two different ways to mark the bid won (or three if you count the non-native case).

Reply:
I mean something like - 

```
const BID_WONS = new WeakMap();

function receiveMessage(ev) {
 // ...
 if !BID_WONS.has(adObject) {
    BID_WONS.set(adObject, true);
    // ... trigger event
 }
 // message handling
}

Reply:
updated